docs/platform-integration/essentials/permissions.md

```diff
@@ -123,78 +132,88 @@ public async Task<PermissionStatus> CheckAndRequestLocationPermission()
     {
         throw new System.NotImplementedException();
     }
 
-    // Requests the user to accept or deny a permission
+    // Requests the user to accept or deny a permission.
     public override Task<PermissionStatus> RequestAsync()
     {
         throw new System.NotImplementedException();
     }
+
+    // Indicates that the requestor should prompt the user as to why the app requires the permission, because the
+    // user has previously denied this permission.
+    public override bool ShouldShowRationale()
+    {
+        throw new NotImplementedException();
+    }
 }
 ```
 
-When implementing a permission in a specific platform, the `BasePlatformPermission` class can be inherited from. This provides additional platform helper methods to automatically check the declarations. This can help when creating custom permissions that do groupings. For example, you can request both Read and Write access to storage on Android using the following custom permission.
+When implementing a permission in a specific platform, the `BasePlatformPermission` class can be inherited from. This class provides extra platform helper methods to automatically check the permission declarations. This helps when creating custom permissions that do groupings, for example requesting both **Read** and **Write** access to storage on Android. The following code example demonstrates requesting **Read** and **Write** storage access:
 
 ```csharp
-public class ReadWriteStoragePermission : Xamarin.Essentials.Permissions.BasePlatformPermission
+public class ReadWriteStoragePerms: Microsoft.Maui.Essentials.Permissions.BasePlatformPermission
 {
-    public override (string androidPermission, bool isRuntime)[] RequiredPermissions => new List<(string androidPermission, bool isRuntime)>
-    {
-        (Android.Manifest.Permission.ReadExternalStorage, true),
-        (Android.Manifest.Permission.WriteExternalStorage, true)
-    }.ToArray();
+    public override (string androidPermission, bool isRuntime)[] RequiredPermissions =>
+        new List<(string androidPermission, bool isRuntime)>
+        {
+            (global::Android.Manifest.Permission.ReadExternalStorage, true),
+            (global::Android.Manifest.Permission.WriteExternalStorage, true)
+        }.ToArray();
 }
 ```
 
-Then you can call your new permission from Android project.
+You then check the permission in the same way as any other permission type provided by .NET MAUI:
 
 ```csharp
 await Permissions.RequestAsync<ReadWriteStoragePermission>();
 ```
 
-If you wanted to call this API from your shared code you could create an interface and use a dependency service to register and get the implementation.
+<!-- TODO: Is this still the same? Does shared code just pull in the platform specific code at compile time and thus you don't need an interface as long as you've declared the type in all platforms you're compiling for? (The rest of this section was not processed for this update and needs to be updated)-->
```

Comment:
It's the Xamarin.Forms way of doing things. It will still work in .NET MAUI, in order to support users who are migrating their apps to MAUI. But it's the old pattern. So I recommend a todo/issue to update/replace it with the new approach (using multi-targeting with partial classes/methods).